### PR TITLE
hello: update to 2.11

### PR DESCRIPTION
**Pull request by aeiouaeiouaeiouaeiouaeiouaeiou**
###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.11.6 15G22010 x86_64
Xcode 8.2.1 8C1002

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by aeiouaeiouaeiouaeiouaeiouaeiou** [comment]
```
   Undefined symbols for architecture x86_64:
    "_iconv", referenced from:
        _mem_cd_iconv in libhello.a(striconv.o)
        _str_cd_iconv in libhello.a(striconv.o)
       (maybe you meant: _str_cd_iconv, _mem_cd_iconv , _xstr_cd_iconv , _xstr_iconv , _str_iconv , _xmem_cd_iconv )
    "_iconv_close", referenced from:
        _str_iconv in libhello.a(striconv.o)
    "_iconv_open", referenced from:
        _str_iconv in libhello.a(striconv.o)
  ld: symbol(s) not found for architecture x86_64
  clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
It's weird that such a simple utility is now stuck on this problem.

**Comment by pmetzger** [comment]
Did it build on your machine?

**Comment by aeiouaeiouaeiouaeiouaeiouaeiou** [comment]
@pmetzger no, I even tried to build with [`libiconv`](https://ports.macports.org/port/libiconv/details/) dependency and used flags to redirect calls as in [this port](https://github.com/macports/macports-ports/blob/master/sysutils/dosfstools/Portfile#L34).

**Comment by jmroot** [comment]
The proximate cause is that the link command is missing `-liconv`. Discovering why that is may require digging in the Makefiles, since the configure script seemed to correctly decide that that is how it should link with libiconv.

**Comment by ryandesign** [comment]
> I even tried to build with [`libiconv`](https://ports.macports.org/port/libiconv/details/) dependency and used flags to redirect calls as in [this port](https://github.com/macports/macports-ports/blob/master/sysutils/dosfstools/Portfile#L34).

What dosfstools is doing there is extremely wrong and I'll look into fixing it.

Since hello is using libiconv, please do add `depends_lib port:libiconv`.

> The proximate cause is that the link command is missing `-liconv`

> the configure script seemed to correctly decide that that is how it should link with libiconv

Right!

> ```
> checking how to link with libiconv... -liconv
> ```

Presumably whatever variable they've stashed `-liconv` into just needs to be added to the link line.

**Comment by ryandesign** [comment]
Please break this PR into 2 commits: 1 to reformat the whitespace and 1 to update the port. By combining both into one, it's impossible to see what functional changes are being made.

**Comment by pmetzger** [comment]
@ryandesign Just FYI, there's now a setting in Github that hides the whitespace changes and makes reading things like this pretty simple.
image

**Comment by ryandesign** [comment]
That's good, but diffs aren't always viewed on GitHub. It has been our custom for a long time that comprehensive whitespace changes should be in a separate commit.

**Comment by aeiouaeiouaeiouaeiouaeiouaeiou** [comment]
I tried to build with `configure.env LDFLAGS=-liconv` and did it successfully, should I commit this separately?

**Comment by ryandesign** [comment]
You can add it to this PR.

MacPorts already sets some LDFLAGS. You should add to them like this:
```
configure.ldflags-append -liconv
```
Best of all, of course, would be to fix the build system.

File: mail/hello/Portfile [hunk 1]
```diff
@@ -1,21 +1,30 @@
...
-                 sha256 31e066137a962676e89f69d1b65382de95a7ef7d914b8cb956f41ea72e0f516b
-configure.args   --disable-nls \
-                 --infodir=${prefix}/share/info \
-                 --mandir=${prefix}/share/man
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+
+name                hello
+version             2.11
```
Comment (ryandesign) [inline mail/hello/Portfile:6]: The formatting-fix commit shouldn't change the version.

**Comment by ryandesign** [comment]
> Since hello is using libiconv, please do add `depends_lib port:libiconv`.

Doesn't look like you added this dependency yet. I am assuming that, if the libiconv port is installed, hello will link with its library, whereas if the libiconv port is not installed, then it will link with macOS libiconv. To avoid this discrepancy, we should make the port always do one or the other, and using the port seems like the better option.

**Pull request merged**